lib/net/FetchHttpClient.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const {HttpClient, HttpClientResponse} = require('./HttpClient');
+
+/**
+ * HTTP client which uses a `fetch` function to issue requests. This fetch
+ * function is expected to be the Web Fetch API function or an equivalent (such
+ * as the function provided by the node-fetch package).
+ */
+class FetchHttpClient extends HttpClient {
+  constructor(fetchFn) {
+    super();
+    this._fetchFn = fetchFn;
+  }
+
+  /* eslint-disable class-methods-use-this */
```

Comment:
Shall we globally disable this rule? We seem to violate it all the time and I don't really agree with it.

Reply:
Done. Agreed.